r/R/compute.R

```diff
@@ -229,6 +233,25 @@ all.ArrowDatum <- function(..., na.rm = FALSE){
 #' as `x` with the (0-based) indexes into `table`. `is_in()` returns a
 #' `boolean`-type `Array` of the same length as `x` with values indicating
 #' per element of `x` it it is present in `table`.
+#' @examples
+#' # note that the returned value is 0-indexed
+#' cars_tbl <- Table$create(name = rownames(mtcars), mtcars)
+#' match_arrow(Array$create("Mazda RX4 Wag"), cars_tbl$name)
```

Comment:
Would it be helpful to include something like `match_arrow(Scalar$create("Mazda RX4 Wag"), cars_tbl$name)` as well to show that one can use a scalar or an array? It looks like the docs above don't actually mention this, so maybe it's an off-label use that we should avoid? Or it's been added and we should document it...

Reply:
Good catch there - this has been implemented for all ArrowDatum objects.  I also had a read of the docs for the C++ compute function and experimented with different objects, and it appears that the output type == the x input type, so I have documented this as well now.